### PR TITLE
Prevent multiple bug refs per comment to yield incorrect URLs

### PR DESCRIPTION
**Pull request by Martchus**
Related progress issue: poo#11970

### CODE REVIEW

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -305,15 +306,25 @@ my %bugrefs = (
 );
 my %bugurls = reverse %bugrefs;
 
+sub bugref_regex {
+    my $regex = join('|', keys %bugrefs);
+    return qr/(($regex)#(\d+))(?=\s|$)/;
```
Comment (aaannz) [inline lib/OpenQA/Utils.pm]: One thing I noticed that you can't separate bug refs and urls by anything else then space or eol. I imagine comma, semicolon and dot should be also fine.

**Comment by okurz** [comment]
...and colon, maybe more. Maybe it's easier after all to not check for any characters after the digits

**Comment by Martchus** [comment]
What about negative look-up assertion?

File: lib/OpenQA/Utils.pm [hunk 2]
```diff
@@ -305,15 +306,25 @@ my %bugrefs = (
 );
 my %bugurls = reverse %bugrefs;
 
+sub bugref_regex {
+    my $regex = join('|', keys %bugrefs);
+    return qr/(($regex)#(\d+))(?![\w\d]+)/;
```
Comment (aaannz) [inline lib/OpenQA/Utils.pm]: I would rather not put `\d` into the forward lookup. I guess its fine function wise, but it feels superfluous to me. "Accept all digits and then there shouldn't be any word character or digit". Why also matching more than one of these characters?

Comment (Martchus) [inline lib/OpenQA/Utils.pm]: You're right. These were just some leftovers from experimenting with the expression. I fixed it and also the occurrence in `href_to_bugref`.

**Pull request merged**